docs/cudf/source/cudf_polars/streaming_execution.md

```diff
@@ -0,0 +1,90 @@
+# Experimental streaming and multi-GPU options
+
+As well as in-memory execution, obtained by selecting `engine="gpu"`
+when `collect`ing a query, the GPU engine supports streaming execution
+that partitions the data in the query into chunks. To select streaming
+execution, we need to pass an appropriately configured `GPUEngine`
+object into `collect`.
+
+The streaming executors work best when the inputs to your query come
+from parquet files. That is, start with `scan_parquet`, not existing
+Polars `DataFrame`s or CSV files.
+
+````{note}
+Streaming execution is in an experimental state and not all queries
+will run in streaming mode.
+````
+
+## Single GPU streaming
+
+The simplest case, requiring no additional dependencies, is the
+`synchronous` executor. An appropriate engine is:
+```python
+engine = pl.GPUEngine(
+    executor="streaming",
+    executor_options={"scheduler": "synchronous"},
+)
+```
+
+When executed with this engine, any parquet inputs are split into
+"partitions" that are streamed through the query graph. We try and
+pick a good default for the typical partition size (based on the
+amount of GPU memory available), however, it might not be optimal. You
+can configure the execution by providing more options to the executor.
+For example, to configure the max number of rows in each partition:
+``` executor_options={ ..., "max_rows_per_partition": 1_000_000, } ```
```

Comment:
Note that `"max_rows_per_partition"` only works for `DataFrameScan`. For Parquet data, we only consider the `target_partition_size` configuration.  The challenge with `"max_rows_per_partition"` is that we typically don't want to guarantee that a partition will not exceed this row count for on-disk data, because we don't want to take the time to ensure our partitions are reading *exactly* this many rows (or fewer).

Reply:
Updated to use `target_partition_size` in the example and added a bit of text saying `max_rows_per_partition` can be used for in-memory dataframes.